File: src/Entity/Voiture.php
<?php

namespace App\Entity;

use App\Repository\VoitureRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\HttpFoundation\File\UploadedFile;
use Symfony\Component\Validator\Constraints as Assert;

/**
 * @ORM\Entity(repositoryClass=VoitureRepository::class)
 */
class Voiture
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @Assert\NotBlank(message="la Serie doit etre non vide")
     * @Assert\Length(
     *      min = 6,
     *      minMessage=" Entrer une serie au mini de 10 caracteres"
     *
     *     )
     * @ORM\Column(type="string", length=255)
     */
    private $serie;

    /**
     * @Assert\NotBlank(message="la marque doit etre non vide")
     * @Assert\Length(
     *      min = 6,
     *      minMessage=" Entrer une marque au mini de 10 caracteres"
     *
     *     )
     * @ORM\Column(type="string", length=255)
     */
    private $marque;

    /**
     * @Assert\NotBlank(message="le model doit etre non vide")
     * @Assert\Length(
     *      min = 7,
     *      minMessage=" Entrer un model au mini de 10 caracteres"
     *
     *     )
     * @ORM\Column(type="string", length=255)
     */
    private $model;
    /**
     * @Assert\NotBlank(message="la description doit etre non vide")
     * @Assert\Length(
     *      min = 7,
     *      max = 100,
     *      minMessage = "doit etre >=7 ",
     *      maxMessage = "doit etre <=100" )
     * @ORM\Column(type="string", length=1000)
     */
    private $description;
    /**
     * @Assert\NotBlank(message="couleur produit doit etre non vide")
     * @ORM\Column(type="string", length=255)
     */
    private $couleur;



    /**
     * @Assert\NotBlank(message="prix produit doit etre non vide")
     * @Assert\Positive
     * @Assert\Range(
     *      min = 3,
     *      max = 1000,
     *      notInRangeMessage = "le prix doit etre valid",
     *     )
     * @ORM\Column(type="float")
     */
    private $prix;


    /**
     * @ORM\Column(type="string", length=500,nullable=true)
     */
    private $image;


    /**
     * @Assert\File(maxSize="6000000")
     */
    private $file;


    public function getId(): ?int
    {
        return $this->id;
    }

    public function getSerie(): ?string
    {
        return $this->serie;
    }

    public function setSerie(string $serie): self
    {
        $this->serie = $serie;

        return $this;
    }

    public function getMarque(): ?string
    {
        return $this->marque;
    }

    public function setMarque(string $marque): self
    {
        $this->marque = $marque;

        return $this;
    }

    public function getModel(): ?string
    {
        return $this->model;
    }

    public function setModel(string $model): self
    {
        $this->model = $model;

        return $this;
    }

    public function getPrix(): ?float
    {
        return $this->prix;
    }

    public function setPrix(float $prix): self
    {
        $this->prix = $prix;

        return $this;
    }


    public function getdescription(): ?string
    {
        return $this->description;
    }

    public function setdescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getcouleur(): ?string
    {
        return $this->couleur;
    }

    public function setcouleur(string $couleur): self
    {
        $this->couleur = $couleur;

        return $this;
    }
    /**
     * @return mixed
     */
    public function getImage()
    {
        return $this->image;
    }

    /**
     * @param mixed $image
     */
    public function setImage($image): void
    {
        $this->image = $image;
    }

    //UPLOAD IMAGE
    //GeT PUBLIC FOLDER
    public function getPublicFolder() {
        $webPath = $this->get('kernel')->getProjectDir() . '/public/uploads/produit_image';

        return $webPath;


    }

    /**
     * @return UploadedFile
     */
    public function getFile()
    {
        return $this->file;
    }

    /**
     * @param UploadedFile
     */
    public function setFile($file): void
    {
        $this->file = $file;
    }


    //te5o image w tzidha fi dossier produit_image
    public function upload()
    {
        if(null === $this->getFile()) {
            return;
        }

        $this->getFile()->move(
            $this->getPublicFolder(),//destinataire
            $this->getFile()->getClientOriginalName()//esem fichier (image)
        );

        $this->image = $this->getFile()->getClientOriginalName();//

        $this->file = null; // liberation memoire
    }





}

File: GuidiniTravel/src/entities/Voiture.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entities;

import javafx.scene.image.ImageView;

public class Voiture {
    private int id,prix;
    private String marque,model,couleur,description,image,serie;
    private ImageView photo;
 

    public Voiture() {
    }

    public Voiture(int id, String serie,String marque, String model, String description, String couleur, int prix ,String image) {
        this.id = id;
        this.serie = serie;
        this.marque = marque;
        this.model = model;
        this.description = description;
        this.couleur = couleur;
        this.prix= prix;
        this.image= image;
      
    }

 
    public Voiture( String serie,String marque, String model, String description, String couleur, int prix ,String image) {
        
        this.serie = serie;
        this.marque = marque;
        this.model = model;
        this.description = description;
        this.couleur = couleur;
        this.prix= prix;
        this.image= image;
      
    }

   public Voiture(String serie,String marque, String model, String description, String couleur, int prix ,String image, ImageView photo) {
       
        this.photo = photo;
        this.serie = serie;
        this.marque = marque;
        this.model = model;
        this.description = description;
        this.couleur = couleur;
        this.prix= prix;
        this.image= image;
      
    }
    
    
    
    


    

    public Voiture(int i, String serie, String marque, String model, String description, String couleur, String prix, String img) {
        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
    }

    public Voiture(int i, int serie1, String marque, String model, String m, String couleur, int prix1, String img) {
        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
    }

   

 
    
    @Override
    public String toString() {
        return "Voiture{" + "id=" + id + ", serie=" + serie + ", marque=" + marque + ", model =" + model + ", description =" + description  + ", couleur =" + couleur  +", prix =" + prix  + '}';
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }
    
public String getSerie() {
        return serie;
    }

    public void setSerie(String serie) {
        this.serie = serie;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getCouleur() {
        return couleur;
    }

    public void setCouleur(String couleur) {
        this.couleur = couleur;
    }

 

    public String getMarque() {
        return marque;
    }

    public void setMarque(String marque) {
        this.marque = marque;
    }

    public String getModel() {
        return model;
    }

    public void setModel(String model) {
        this.model = model;
    }

   
  
  

    public double getPrix() {
        return prix;
    }

    public void setPrix(int prix) {
        this.prix = prix;
    }

    
    public ImageView getPhoto() {
        return photo;
    }

    public void setPhoto(ImageView photo) {
        this.photo = photo;
    }

    
}
